src/com/reliance/jiocloud/sbs/volumeasyncservice/workflow/deletevolume/tasks/QuotaUpdateTask.java

```
@@ -0,0 +1,53 @@
+package com.reliance.jiocloud.sbs.volumeasyncservice.workflow.deletevolume.tasks;
+
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+import com.reliance.jiocloud.sbs.db.driver.IDataFetch;
+import com.reliance.jiocloud.sbs.volumeasyncservice.Constants;
+import com.reliance.jiocloud.sbs.volumeasyncservice.workflow.BaseTask;
+import com.reliance.jiocloud.sbs.volumeasyncservice.workflow.IWorkflowContext;
+import com.reliance.jiocloud.sbs.volumeasyncservice.workflow.State;
+import com.reliance.jiocloud.sbs.volumeasyncservice.workflow.volume.VolumeContext;
+import com.reliance.jiocloud.sbs.volumestorage.SbsWorkflowException;
+
+/**
+ * Task to release the quota for an user.
+ * 
+ * @author souvik.ray
+ */
+public class QuotaUpdateTask extends BaseTask {
+    private static final Logger logger = LoggerFactory.getLogger(QuotaUpdateTask.class);
+    private IDataFetch datafetch;
+    private State workflowCompleted;
+    private State workflowFailed;
+
+    public QuotaUpdateTask(IDataFetch datafetch, State currentState, State workflowCompleted, State workflowFailed) {
+        this.datafetch = datafetch;
+        this.workflowCompleted = workflowCompleted;
+        this.workflowFailed = workflowFailed;
+    }
+
+    @Override
+    public State execute(IWorkflowContext context) throws SbsWorkflowException {
+        VolumeContext volumeContext = (VolumeContext) context;
+        String projectId = volumeContext.getAttribute(Constants.PROJECT_ID);
+        int size = Integer.parseInt(volumeContext.getAttribute(Constants.SIZE));
+        try {
+            Map<String, Integer> quotaDeltas = Maps.newHashMap();
+            quotaDeltas.put("volumes", -1);
+            quotaDeltas.put("gigabytes", -size);
+            datafetch.updateQuotaUsages(projectId,
+                            quotaDeltas);
+            return workflowCompleted;
+        } catch (Exception e) {
```

Catching all exceptions at this deep in the application is 'asking-for-trouble', IMO. It's ok to catch all exceptions at the application entry point provided the application understands that it can recover from that situation. But in the thick of the application, being conservative with catching exceptions is the recommended approach.